packages/adblocker/src/engine/engine.ts

```diff
@@ -489,11 +526,33 @@ export default class FilterEngine extends EventEmitter<
   public update({
     newNetworkFilters = [],
     newCosmeticFilters = [],
+    newPreprocessors = [],
     removedCosmeticFilters = [],
     removedNetworkFilters = [],
-  }: Partial<IListDiff>): boolean {
+    removedPreprocessors = [],
+    env = new Env(),
```

Comment:
I think `env` could be a second argument instead of passing it as part of the diff (first argument).

Reply:
@remusao should this apply to `#updateFromDiff` and to `PreprocessorBucker.update` as well?